src/server/services/image.service.ts

```diff
@@ -2011,3 +2012,80 @@ export const getEntityCoverImage = async ({
     tags: tagsVar?.filter((x) => x.imageId === i.id),
   }));
 };
+
+export const updateEntityImages = async ({
+  tx,
+  entityId,
+  entityType,
+  images,
+  userId,
+}: {
+  tx: Prisma.TransactionClient;
+  entityId: number;
+  entityType: string;
+  images: ImageUploadProps[];
+  userId: number;
+}) => {
+  const connections = await tx.imageConnection.findMany({
```

Comment:
One small trick I've been using to be able to reuse this outside a transaction is the following
```suggestion
}) => {
  const dbClient = tx ?? dbWrite;
  const connections = await dbClient.imageConnection.findMany({
```
This way it's not mandatory to call this inside a prisma transaction. Make sure you make `tx` optional